autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -254,11 +273,29 @@ def on_epoch_end(self, X: Dict[str, Any], epoch: int) -> bool:
         """
         return False
 
+    def _scheduler_step(
+        self,
+        step_interval: StepIntervalUnit,
+        loss: Optional[float] = None
+    ) -> None:
+
+        if self.step_interval != step_interval:
+            return
+
+        if not self.scheduler:  # skip if no scheduler defined
+            return
+
+        try:
```

Comment:
I think it was better before. why do you want to catch an exception and then handle it?

Reply:
because we do not know what kind of `lr_scheduler` we add in the future.
In reality, I would like to have got the arguments for each function, but the simplest way I can achieve this was to catch by TypeError.

Reply:
Okay, I see.  Could you create an issue for this? So we don't forget to update this. Moreover, as we allow the users to add an LR scheduler, we should think of a neat way to pass this info.

Reply:
Here, I posted
https://github.com/automl/Auto-PyTorch/issues/270